### ISSUE TITLE
Understanding db using

### ISSUE THREAD

**Issue by lud4ik** [tailhook/vagga#34]
For example, I have the following setup:
```
  ubuntu-mongo:
    setup:
    - !Container ubuntu
    - !Install [ca-certificates]
    - !Sh |
        sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10
        echo "deb http://repo.mongodb.org/apt/ubuntu "$(lsb_release -sc)"/mongodb-org/3.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.0.list
        sudo apt-get update
        sudo apt-get install -y mongodb-org


  mongo: !Command
    description: Mongo server
    container: ubuntu-mongo
    run: |
        mongo
```

When I start it, it shows databases of mongo installed locally. I don't understand why. Is it my error, or I have to care about using proper names not to coincide with local?
With mysql I'll have the same?

Output
```
lud4ik@lud4ik-linux:~/main/mail/smart-delivery$ vagga mongo
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty InRelease
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates InRelease
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security InRelease
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty Release.gpg
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates Release.gpg
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security Release.gpg
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty Release  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates Release
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security Release         
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main Sources             
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted Sources
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main amd64 Packages      
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/universe amd64 Packages  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main Translation-en      
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/universe Translation-en  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/main Sources     
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/restricted Sources
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/main amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/restricted amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/universe amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/main Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/restricted Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/universe Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/main Sources    
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/restricted Sources
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/main amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/restricted amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/universe amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/main Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/restricted Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/universe Translation-en
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main Translation-en_US   
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted Translation-en_US
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/universe Translation-en_US
Reading package lists... Done                                                  
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  openssl
The following NEW packages will be installed:
  ca-certificates openssl
0 upgraded, 2 newly installed, 0 to remove and 4 not upgraded.
Need to get 0 B/678 kB of archives.
After this operation, 1,392 kB of additional disk space will be used.
Preconfiguring packages ...
E: Can not write log (Is /dev/pts mounted?) - openpty (13: Permission denied)
Selecting previously unselected package openssl.
(Reading database ... 11527 files and directories currently installed.)
Preparing to unpack .../openssl_1.0.1f-1ubuntu2.15_amd64.deb ...
Unpacking openssl (1.0.1f-1ubuntu2.15) ...
Selecting previously unselected package ca-certificates.
Preparing to unpack .../ca-certificates_20141019ubuntu0.14.04.1_all.deb ...
Unpacking ca-certificates (20141019ubuntu0.14.04.1) ...
Setting up openssl (1.0.1f-1ubuntu2.15) ...
Setting up ca-certificates (20141019ubuntu0.14.04.1) ...
Processing triggers for ca-certificates (20141019ubuntu0.14.04.1) ...
Updating certificates in /etc/ssl/certs... 173 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d....done.
sudo: unable to resolve host lud4ik-linux
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --homedir /tmp/tmp.wPG6ImLL79 --no-auto-check-trustdb --trust-model always --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10
gpg: requesting key 7F0CEB10 from hkp server keyserver.ubuntu.com
gpg: key 7F0CEB10: public key "Richard Kreuter <richard@10gen.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
sudo: unable to resolve host lud4ik-linux
deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.0 multiverse
sudo: unable to resolve host lud4ik-linux
Ign http://repo.mongodb.org trusty/mongodb-org/3.0 InRelease
Get:1 http://repo.mongodb.org trusty/mongodb-org/3.0 Release.gpg [490 B]
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty InRelease                
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates InRelease
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security InRelease
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty Release.gpg
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates Release.gpg      
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security Release.gpg     
Get:2 http://repo.mongodb.org trusty/mongodb-org/3.0 Release [2,119 B]
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty Release                  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates Release  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security Release 
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main Sources
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted Sources       
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main amd64 Packages      
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted amd64 Packages
Get:3 http://repo.mongodb.org trusty/mongodb-org/3.0/multiverse amd64 Packages [4,768 B]
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/universe amd64 Packages  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main Translation-en      
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/universe Translation-en  
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/main Sources     
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/restricted Sources
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/main amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/restricted amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/universe amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/main Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/restricted Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-updates/universe Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/main Sources    
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/restricted Sources
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/main amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/restricted amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/universe amd64 Packages
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/main Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/restricted Translation-en
Hit http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty-security/universe Translation-en
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/main Translation-en_US   
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/restricted Translation-en_US
Ign http://ubuntu-mirror.neocom.org.ua/ubuntu/ trusty/universe Translation-en_US
Ign http://repo.mongodb.org trusty/mongodb-org/3.0/multiverse Translation-en_US
Ign http://repo.mongodb.org trusty/mongodb-org/3.0/multiverse Translation-en
Fetched 7,377 B in 1s (5,021 B/s)
Reading package lists... Done
sudo: unable to resolve host lud4ik-linux
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  mongodb-org-mongos mongodb-org-server mongodb-org-shell mongodb-org-tools
The following NEW packages will be installed:
  mongodb-org mongodb-org-mongos mongodb-org-server mongodb-org-shell
  mongodb-org-tools
0 upgraded, 5 newly installed, 0 to remove and 4 not upgraded.
Need to get 50.7 MB of archives.
After this operation, 157 MB of additional disk space will be used.
Get:1 http://repo.mongodb.org/apt/ubuntu/ trusty/mongodb-org/3.0/multiverse mongodb-org-shell amd64 3.0.4 [4,245 kB]
Get:2 http://repo.mongodb.org/apt/ubuntu/ trusty/mongodb-org/3.0/multiverse mongodb-org-server amd64 3.0.4 [8,607 kB]
Get:3 http://repo.mongodb.org/apt/ubuntu/ trusty/mongodb-org/3.0/multiverse mongodb-org-mongos amd64 3.0.4 [4,029 kB]
Get:4 http://repo.mongodb.org/apt/ubuntu/ trusty/mongodb-org/3.0/multiverse mongodb-org-tools amd64 3.0.4 [33.8 MB]
Get:5 http://repo.mongodb.org/apt/ubuntu/ trusty/mongodb-org/3.0/multiverse mongodb-org amd64 3.0.4 [3,616 B]
Fetched 50.7 MB in 8s (6,044 kB/s)                                             
debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
debconf: falling back to frontend: Readline
E: Can not write log (Is /dev/pts mounted?) - openpty (13: Permission denied)
Selecting previously unselected package mongodb-org-shell.
(Reading database ... 11821 files and directories currently installed.)
Preparing to unpack .../mongodb-org-shell_3.0.4_amd64.deb ...
Unpacking mongodb-org-shell (3.0.4) ...
Selecting previously unselected package mongodb-org-server.
Preparing to unpack .../mongodb-org-server_3.0.4_amd64.deb ...
Unpacking mongodb-org-server (3.0.4) ...
Selecting previously unselected package mongodb-org-mongos.
Preparing to unpack .../mongodb-org-mongos_3.0.4_amd64.deb ...
Unpacking mongodb-org-mongos (3.0.4) ...
Selecting previously unselected package mongodb-org-tools.
Preparing to unpack .../mongodb-org-tools_3.0.4_amd64.deb ...
Unpacking mongodb-org-tools (3.0.4) ...
Selecting previously unselected package mongodb-org.
Preparing to unpack .../mongodb-org_3.0.4_amd64.deb ...
Unpacking mongodb-org (3.0.4) ...
Processing triggers for ureadahead (0.100.0-16) ...
Setting up mongodb-org-shell (3.0.4) ...
Setting up mongodb-org-server (3.0.4) ...
Adding system user `mongodb' (UID 102) ...
Adding new user `mongodb' (UID 102) with group `nogroup' ...
Not creating home directory `/home/mongodb'.
Adding group `mongodb' (GID 105) ...
Done.
Adding user `mongodb' to group `mongodb' ...
Adding user mongodb to group mongodb
Done.
initctl: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
invoke-rc.d: policy-rc.d denied execution of start.
Setting up mongodb-org-mongos (3.0.4) ...
Setting up mongodb-org-tools (3.0.4) ...
Processing triggers for ureadahead (0.100.0-16) ...
Setting up mongodb-org (3.0.4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 4 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 4 not upgraded.
MongoDB shell version: 3.0.4
connecting to: test
```

**Comment by tailhook** [comment]
> When I start it, it shows databases of mongo installed locally

IIRC "mongo" command is mongo client. Probably you have mongodb running on your host system, and client running inside vagga which connects to it. To run mongodb inside container you must run 
"mongod" (is it right name for server binary?) by vagga itself. It's not started as a system service, so you probably want to run it using `!Supervise` command:
```
commands:
  run_app_and_mongo:
    children:
      mongo_db_server:
        run: mongod
      mongo_client:
        run: mongo
```
You will also need to shut down system mongo service or change port of mongod to avoid conflicts. Also you are supposed to configure mongodb to store files somewhere in `/work` or in tmpfs (with either `!Tmpfs` or `BindRW` volume, or by configuring mongo itself).

**Comment by lud4ik** [comment]
Thanks. How to connect from app container to mongo in other container? Set IP for first?

**Comment by tailhook** [comment]
No, by default apps use host network namespace. I.e. they bind to the same ports and have same localhost (127.0.0.1), so you can connect programs inside and ouside the containers just like there are no containers at all.

I.e. your problem is running mongo on host and not running it inside container, not network connectivity.

**Comment by lud4ik** [comment]
```
Setting up mongodb-org-server (3.0.4) ...
Adding system user `mongodb' (UID 102) ...
Adding new user `mongodb' (UID 102) with group `nogroup' ...
Not creating home directory `/home/mongodb'.
Adding group `mongodb' (GID 105) ...
Done.
Adding user `mongodb' to group `mongodb' ...
Adding user mongodb to group mongodb
Done.
initctl: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
invoke-rc.d: policy-rc.d denied execution of start.
Setting up mongodb-org-mongos (3.0.4) ...
Setting up mongodb-org-tools (3.0.4) ...
Processing triggers for ureadahead (0.100.0-16) ...
Setting up mongodb-org (3.0.4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
sudo: unable to resolve host luda-K43SJ
mongodb: unrecognized service
MongoDB shell version: 3.0.4
connecting to: test
2015-06-16T19:02:13.683+0000 W NETWORK  Failed to connect to 127.0.0.1:27017, reason: errno:111 Connection refused
2015-06-16T19:02:13.685+0000 E QUERY    Error: couldn't connect to server 127.0.0.1:27017 (127.0.0.1), connection attempt failed
    at connect (src/mongo/shell/mongo.js:179:14)
    at (connect):1:6 at src/mongo/shell/mongo.js:179

```

**Comment by lud4ik** [comment]
хосты нада прописать?

**Comment by tailhook** [comment]
Please show me config. I guess you either did not start "mongod" or you start them simultaneously and client start much faster than server. You can try to add sleep before starting client:
```
mongo_client:
  run: |
    sleep 5
    mongo
```

**Comment by lud4ik** [comment]
I have one related question. I try to start my app that need to connect to mysql in vagrant:
```
path = root@tcp(crm.dev:3306)/db_crm?autocommit=true&charset=utf8
2015-06-26 08:08:21.18 main - [CRITICAL] crmchatserver.go:175 :: Error connecting to database:Error 1049: Unknown database 'db_crm'
```
Database exists and from local machine it connects, but from container get this err. Do I need some proxing?

**Comment by lud4ik** [comment]
Ah, sorry I guess I should add it to /etc/hosts.

**Comment by tailhook** [comment]
Well, yes it seems vagga copies `/etc/resolv.conf` but does not copy `/etc/hosts` to container. I don't remember if it was intentional, but please create a separate issue for this. I'll look into it right after porting vagga to rust-1.x

As a temporary workaround you may either write `/etc/hosts` when building container, or use IP address.

**Comment by lud4ik** [comment]
ok thanks!

**Comment by lud4ik** [comment]
Cfg for mongo
```
commands:

  mongo: !Command
    description: Mongo server
    container: ubuntu-mongo
    run: |
        sudo service mongod start

containers:

  ubuntu:
    setup:
    - !Ubuntu trusty
    - !UbuntuUniverse ~
    - !EnsureDir /var/lib/apt/mirrors

  ubuntu-mongo:
    setup:
    - !Container ubuntu
    - !Install [ca-certificates]
    - !Sh |
        sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10
        echo "deb http://repo.mongodb.org/apt/ubuntu "$(lsb_release -sc)"/mongodb-org/3.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.0.list
        sudo apt-get update
        sudo apt-get install -y mongodb-org
```
Err
```
Rather than invoking init scripts through /etc/init.d, use the service(8)
utility, e.g. service mongod start
initctl: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.

Since the script you are attempting to invoke has been converted to an
Upstart job, you may also use the start(8) utility, e.g. start mongod
start: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
```
Am I doing smth wrong?
